### PR TITLE
Allow setting team presets for external matchmaking

### PR DESCRIPTION
**Pull request by Brutus5000**

### CODE REVIEW

File: faf-java-server-app/src/main/java/com/faforever/server/matchmaker/MatchMakerService.java [hunk 1]
```diff
@@ -281,13 +282,13 @@ private void notifyPlayers(MatchMakerSearch search) {
...
-    log.debug("Creating match '{}' with '{}' participants on map '{}'", title, participants.size(), map);
+    log.debug("Creating match '{}' with '{}' presetParticipants on map '{}'", title, presetParticipants.size(), map);
 
-    List<Player> players = participants.stream()
-      .map(matchParticipant -> playerService.getOnlinePlayer(matchParticipant.getId())
-        .orElseThrow(() -> new RequestException(requestId, ErrorCode.PLAYER_NOT_AVAILABLE_FOR_MATCHMAKING_OFFLINE, matchParticipant.getId())))
+    List<Player> players = presetParticipants.stream()
+      .map(gameParticipant -> playerService.getOnlinePlayer(gameParticipant.getId())
+        .orElseThrow(() -> new RequestException(requestId, ErrorCode.PLAYER_NOT_AVAILABLE_FOR_MATCHMAKING_OFFLINE, gameParticipant.getId())))
       .peek(player -> Requests.verify(player.getCurrentGame() == null, requestId, ErrorCode.PLAYER_NOT_AVAILABLE_FOR_MATCHMAKING_OFFLINE, player))
```
Comment (1-alex98) [inline faf-java-server-app/src/main/java/com/faforever/server/matchmaker/MatchMakerService.java:292]: I know that is not part of the change, but why is it PLAYER_NOT_AVAILABLE_FOR_MATCHMAKING_OFFLINE, it seems to be rather PLAYER_ALREADY in a game right?

**Review by micheljung — APPROVED** [review]
Solid!

File: faf-java-server-app/src/main/java/com/faforever/server/matchmaker/MatchMakerService.java (hunk 1 above)
Comment (micheljung) [reply faf-java-server-app/src/main/java/com/faforever/server/matchmaker/MatchMakerService.java:292]: if `getOnlinePlayer()` doesn't return the player the he's not online. sounds correct?

Comment (1-alex98) [reply faf-java-server-app/src/main/java/com/faforever/server/matchmaker/MatchMakerService.java:292]: I mean this line `.peek(player -> Requests.verify(player.getCurrentGame() == null, requestId, ErrorCode.PLAYER_NOT_AVAILABLE_FOR_MATCHMAKING_OFFLINE, player))`

Comment (micheljung) [reply faf-java-server-app/src/main/java/com/faforever/server/matchmaker/MatchMakerService.java:292]: Ah, missed that one on my phone. You're right. #100

**Pull request merged**